tools/rpctypegen/core/src/lib.rs

```diff
@@ -114,8 +116,9 @@ mod tests {
         })
         .unwrap();
         parse_error_type(&mut schema, &error_type);
-        let expected: BTreeMap<String, ErrorType> = serde_json::from_str(
-            r#"
+        let expected: BTreeMap<String, ErrorType> =
+            serde_json::from_str::<DeserializedWithDebug<_>>(
```

Comment:
Similarly, keys won't be ignored when deserializing to things like `BTreeMap`. I believe deserialization will just fail if the type is wrong or something.